src/test/java/io/cdap/plugin/servicenow/sink/ServiceNowSinkConfigTest.java

```diff
@@ -0,0 +1,585 @@
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testValidateClientIdNull() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                 .setClientId(null).build(), collector);
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+      Assert.fail("Exception is not thrown with valid client id");
+    } catch (ValidationException e) {
+      Assert.assertEquals(ServiceNowConstants.PROPERTY_CLIENT_ID, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testValidateClientSecretNull() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                 .setClientSecret(null).build(), collector);
+
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+      Assert.fail("Exception is not thrown with valid client secret");
+    } catch (ValidationException e) {
+      Assert.assertEquals(ServiceNowConstants.PROPERTY_CLIENT_SECRET, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testValidateApiEndpointNull() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                 .setRestApiEndpoint(null).build(), collector);
+
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+      Assert.fail("Exception is not thrown with valid Api Endpoint");
+    } catch (ValidationException e) {
+      Assert.assertEquals(ServiceNowConstants.PROPERTY_API_ENDPOINT, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testValidateUserNull() {
+    MockFailureCollector collector = new MockFailureCollector();
+    ServiceNowSinkConfig config = withServiceNowValidationMock(ServiceNowSinkConfigHelper.newConfigBuilder()
+                                                                 .setUser(null).build(), collector);
+
+    try {
+      config.validate(collector);
+      collector.getOrThrowException();
+      Assert.fail("Exception is not thrown with valid User");
+    } catch (ValidationException e) {
+      Assert.assertEquals(ServiceNowConstants.PROPERTY_USER, e.getFailures().get(0).getCauses().get(0)
+        .getAttribute(CauseAttributes.STAGE_CONFIG));
+    }
+  }
+
+  @Test
+  public void testValidatePasswordNull() {
```

Comment:
Can we combine all the above null checks in one method?

Reply:
Can't be done as assertions are different for each failure.